contracts/leaser/src/migrate.rs

```diff
@@ -2,101 +2,305 @@ use lease::api::MigrateMsg;
+            );
+        }
+        {
+            let exp = add_expected(MigrationResult::default(), &lease1, new_code);
+            let exp = add_expected(exp, &lease21, new_code);
+            let mut exp = add_expected(exp, &lease22, new_code);
+            exp.next_customer = Some(customer_addr3);
+            assert_eq!(
+                Ok(exp),
+                super::migrate_leases(test_customers(), new_code, 3)
+            );
+        }
+        {
+            let exp = add_expected(MigrationResult::default(), &lease1, new_code);
+            let exp = add_expected(exp, &lease21, new_code);
+            let exp = add_expected(exp, &lease22, new_code);
+            let mut exp = add_expected(exp, &lease3, new_code);
+            exp.next_customer = Some(customer_addr4.clone());
+            assert_eq!(
+                Ok(exp),
+                super::migrate_leases(test_customers(), new_code, 4)
+            );
+        }
+        {
+            let exp = add_expected(MigrationResult::default(), &lease1, new_code);
+            let exp = add_expected(exp, &lease21, new_code);
+            let exp = add_expected(exp, &lease22, new_code);
+            let mut exp = add_expected(exp, &lease3, new_code);
+            exp.next_customer = Some(customer_addr4);
+            assert_eq!(
+                Ok(exp),
+                super::migrate_leases(test_customers(), new_code, 5)
+            );
+        }
+        {
+            let exp = add_expected(MigrationResult::default(), &lease1, new_code);
+            let exp = add_expected(exp, &lease21, new_code);
+            let exp = add_expected(exp, &lease22, new_code);
+            let exp = add_expected(exp, &lease3, new_code);
+            let exp = add_expected(exp, &lease41, new_code);
+            let exp = add_expected(exp, &lease42, new_code);
+            let mut exp = add_expected(exp, &lease43, new_code);
+            exp.next_customer = None;
+            assert_eq!(
+                Ok(exp),
+                super::migrate_leases(test_customers(), new_code, 7)
+            );
+        }
     }
 
     #[test]
     fn err_leases() {
         let new_code = 242;
+        let lease1 = Addr::unchecked("lease11");
+        let lease2 = Addr::unchecked("lease12");
+        let lease3 = Addr::unchecked("lease13");
+        let cust1 = Customer::from(
+            Addr::unchecked("customer1"),
+            vec![lease1, lease2, lease3].into_iter(),
+        );
         let err = "testing error";
 
-        let no_leases = vec![
-            Ok(Addr::unchecked("fsdffg")),
+        let customers = vec![
+            Ok(cust1),
             Err(ContractError::ParseError { err: err.into() }),
-            Ok(Addr::unchecked("2424")),
         ];
         assert_eq!(
             Err(ContractError::ParseError { err: err.into() }),
-            super::migrate_leases(no_leases.into_iter(), new_code)
+            super::migrate_leases(customers.into_iter(), new_code, 3)
         );
```

Comment:
Expected expression is supposed to be the second argument to the macro.
```suggestion
        assert_eq!(
            super::migrate_leases(customers.into_iter(), new_code, 3),
            Err(ContractError::ParseError { err: err.into() })
        );
```